src/uct/tcp/tcp_ep.c

```diff
@@ -2067,17 +2086,28 @@ ucs_status_t uct_tcp_ep_flush(uct_ep_h tl_ep, unsigned flags,
     }
 
     status = uct_tcp_ep_check_tx_res(ep);
-    if (status == UCS_ERR_NO_RESOURCE) {
-        UCT_TL_EP_STAT_FLUSH_WAIT(&ep->super);
-        return UCS_ERR_NO_RESOURCE;
+    if (ucs_unlikely(status != UCS_OK)) {
+        return status;
     }
 
-    if (ep->flags & UCT_TCP_EP_FLAG_PUT_TX_WAITING_ACK) {
-        status = uct_tcp_ep_put_comp_add(ep, comp, ep->tx.put_sn);
+    if (ep->rx.last_acked_sn != ep->tx.sn) {
```

Comment:
can we remove UCT_TCP_EP_FLAG_HAS_OPS_NO_FLUSH flag when receiving ACK?

Reply:
no, the following case won't work then:
```
do {
    uct_worker_progress();
    status = uct_ep_flush();
} while (status == UCS_INPROGRESS);
```